frontend/javascripts/admin/dataset/composition_wizard/04_configure_new_dataset.tsx

```diff
@@ -0,0 +1,285 @@
+
+const FormItem = Form.Item;
+
+export function ConfigureNewDataset(props: WizardComponentProps) {
+  const formRef = React.useRef<FormInstance<any>>(null);
+
+  const onPrev = () => {
+    props.setWizardContext((oldContext) => ({
+      ...oldContext,
+      currentWizardStep: "SelectDatasets",
+    }));
+  };
+
+  const [isLoading, setIsLoading] = useState(false);
+  const activeUser = useSelector((state: OxalisState) => state.activeUser);
+  const isDatasetManagerOrAdmin = Utils.isUserAdminOrDatasetManager(activeUser);
+  const [form] = Form.useForm();
+  const [selectedTeams, setSelectedTeams] = useState<APITeam | Array<APITeam>>([]);
+
+  const { wizardContext } = props;
+  const linkedDatasets = wizardContext.datasets;
+
+  const onRemoveLayer = (layer: LayerLink) => {
+    const oldLayers = form.getFieldValue(["layers"]);
+    const newLayers = oldLayers.filter((existingLayer: LayerLink) => existingLayer !== layer);
+    form.setFieldsValue({ layers: newLayers });
+  };
+
+  const handleTransformImport = async (sourcePoints: Vector3[], targetPoints: Vector3[]) => {
+    const datasets = linkedDatasets;
+    const transformationArr =
+      sourcePoints.length > 0 && targetPoints.length > 0
+        ? [
+            {
+              type: "affine" as const,
+              matrix: flatToNestedMatrix(estimateAffineMatrix4x4(sourcePoints, targetPoints)),
+            },
+          ]
+        : [];
+
+    const newLinks: LayerLink[] = (
+      _.flatMap(datasets, (dataset) =>
+        dataset.dataSource.dataLayers.map((layer) => [dataset, layer]),
+      ) as [APIDataset, APIDataLayer][]
+    ).map(
+      ([dataset, dataLayer]): LayerLink => ({
+        datasetId: {
+          owningOrganization: dataset.owningOrganization,
+          name: dataset.name,
+        },
+        sourceName: dataLayer.name,
+        newName: dataLayer.name,
+        transformations: dataset === datasets[0] ? transformationArr : [],
+      }),
+    );
+    form.setFieldsValue({ layers: newLinks });
+  };
+
+  useEffect(() => {
+    handleTransformImport(wizardContext.sourcePoints, wizardContext.targetPoints);
+  }, []);
+
+  const handleSubmit = async () => {
+    if (activeUser == null) {
+      throw new Error("Cannot upload dataset without being logged in.");
+    }
+    const layers = form.getFieldValue(["layers"]);
+
+    const uploadableDatastores = props.datastores.filter((datastore) => datastore.allowsUpload);
+    const datastoreToUse = uploadableDatastores[0];
+    if (!datastoreToUse) {
+      Toast.error("Could not find datastore that allows uploading.");
+      return;
+    }
+
+    const newDatasetName = form.getFieldValue(["name"]);
+    setIsLoading(true);
+    try {
+      await createDatasetComposition(datastoreToUse.url, {
+        newDatasetName,
+        targetFolderId: form.getFieldValue(["targetFolderId"]),
+        organizationName: "sample_organization",
+        scale: linkedDatasets[1].dataSource.scale,
```

Comment:
From this line and line 76 I get that dataset[0] is transformed to the coordinate system of dataset[1]. Should this be pointed out somewhere or do you think it's irrelevant?

Reply:
Good point. I added this info into the "Upload files" view as this is the place where the user might want to influence this be re-ordering the files (or the contents).